boilerplate/app/screens/WelcomeScreen.tsx

```diff
@@ -13,11 +14,25 @@ interface WelcomeScreenProps extends AppStackScreenProps<"Welcome"> {}
 
 export const WelcomeScreen = observer(function WelcomeScreen(props: WelcomeScreenProps) {
   const { navigation } = props
+  const {
+    authenticationStore: { setAuthToken },
+  } = useStores()
 
   function goNext() {
     navigation.navigate("Demo", { screen: "DemoComponents" })
   }
 
+  function logout() {
+    setAuthToken(undefined)
+  }
+
+  useLayoutEffect(() => {
+    navigation.setOptions({
+      headerShown: true,
+      header: () => <Header rightTx="welcomeScreen.headerRight" onRightPress={logout} />,
+    })
+  }, [])
```

Comment:
shows how to add the Header properly to the navigation, rather than just using it inside the component.